source/draw/PaintEngine.ooc

```diff
@@ -0,0 +1,99 @@
+//
+// Copyright (c) 2011-2014 Simon Mika <simon@mika.se>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use ooc-base
+use ooc-collections
+use ooc-math
+import math
+import Color
+
+Pen: cover {
+	color: ColorBgra { get set }
+	alpha ::= this color alpha
+	alphaAsFloat ::= this alpha as Float / 255.0f
+	init: func@ (=color)
+	init: func@ ~withBgr (colorBgr: ColorBgr) {
+		this color = colorBgr toBgra()
+	}
+	setApha: func@ (alpha: UInt8) {
+		this color = ColorBgra new(this color toBgr(), alpha)
```

Comment:
Hm, probably because `this color` is a property and not a variable... Oh well, I guess it doesn't matter.